test/e2e/helm/cluster_test.go

```diff
@@ -232,7 +232,7 @@ var _ = Describe("Running Helm projects", func() {
 			cmdOpts := []string{
 				"run", "curl", "--image=curlimages/curl:7.68.0", "--restart=OnFailure", "--",
 				"curl", "-v", "-k", "-H", fmt.Sprintf(`Authorization: Bearer %s`, token),
-				fmt.Sprintf("https://%s-controller-manager-metrics-service.%s.svc:8443/metrics", tc.ProjectName, tc.Kubectl.Namespace),
+				fmt.Sprintf("http://%s-controller-manager-metrics-service.%s.svc:8443/metrics", tc.ProjectName, tc.Kubectl.Namespace),
```

Comment:
IIRC the reason for not scaffolding cert-manager in Helm operator was because webhook scaffolding wasn't supported either. I'm not sure if we'd like to scaffold cert-manager config, or just leave it unsecured and add documentation on either using network policy or tls+cert-manager would be a better path (https://book.kubebuilder.io/reference/metrics.html?highlight=metr#metrics-protection)

Reply:
The default is true for `go` scaffold stuff, but here in helm the default is `false` still since this is outside the path of `kubebuilder`. To provide backwards and forward comparability, since helm (and ansible) a user can't adjust the `helm` binary like they can with go, we'll have to add a new flag for helm (the binary) and to also scaffold out a `patch` file to include the `--metrics-secure` flag.

Reply:
reverted this back after other needed changes were added to helm scaffolding.